test/test_torch.py

```diff
@@ -85,14 +85,20 @@ def test_horovod_reinit(self):
 
         hvd.init()
         rank, size = hvd.rank(), hvd.size()
-
         hvd.shutdown()
         hvd.init()
         rank2, size2 = hvd.rank(), hvd.size()
 
         assert rank == rank2
         assert size == size2
 
+    def test_horovod_is_initialized(self):
```

Comment:
I suspect that the order of the tests is causing the failure due to state carryover between tests.  Are you able to reproduce this locally?  What if you moved this test to the bottom of this file?

Reply:
Yes, you are right, the problem is that the test left hvd in a shutdown state. And similarly to `test_horovod_reinit` we need to shutdown and reinit only if gloo is used. I'll send a fix now.